packages/api/src/controllers/stream.ts

```diff
@@ -626,7 +632,8 @@ app.get("/sessions/:parentId", authorizer({}), async (req, res) => {
   if (
     !stream ||
     stream.deleted ||
-    (stream.userId !== req.user.id && !req.isUIAdmin)
+    (stream.userId !== req.user.id && !req.isUIAdmin) ||
+    stream.projectId !== req.project?.id
```

Comment:
Aren't we missing some coalesce logic here? If the stream has an empty project id, we might compare `"" !== undefined` here which is true even though they should match. This is why I had suggested a similar logic to what we have in the SQL filters:
```suggestion
    stream.projectId ?? "" !== req.project?.id ?? ""
```
(applies to other similar checks in the PR)

Unless you think we will never have empty strings saved in the `projectId` in the DB, then we might be safe just comparing them raw everywhere.

Reply:
Fixed